### PR TITLE
proto: add analyze proto

### PR DESCRIPTION
**Pull request by lamxTyler**
This pr implements the proto for analyze push down.
It seems that my generator will change lots of other files, so could some one helps me generate the code？

PTAL @shenli  @hanfei1991  @coocood  @winoros

### CODE REVIEW

File: proto/analyze.proto [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+message AnalyzeReq {
+    optional AnalyzeType tp = 1 [(gogoproto.nullable) = false];
+    optional uint64 start_ts = 2 [(gogoproto.nullable) = false];
+    optional uint64 flags = 3 [(gogoproto.nullable) = false];
+    optional int64 time_zone_offset = 4 [(gogoproto.nullable) = false];
+    optional AnalyzeIndexReq idx_req = 5;
+    optional AnalyzeColumnsReq col_req = 6;
+}
+
+message AnalyzeIndexReq {
```
Comment (hanfei1991) [inline proto/analyze.proto]: we should estimate ndv for index req ?

Comment (lamxTyler) [reply proto/analyze.proto]: It is in the histogram.

File: proto/analyze.proto [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    optional int64 sample_size = 2 [(gogoproto.nullable) = false];
+
+    // sketch_size is the max sketch size.
+    optional int64 sketch_size = 3 [(gogoproto.nullable) = false];
+
+    // columns_info is the info of all the columns that needs to be analyzed.
+    repeated ColumnInfo columns_info = 4;
+}
+
+message AnalyzeColRespose {
```
Comment (hanfei1991) [inline proto/analyze.proto]: response

File: proto/analyze.proto (hunk 1 above)
Comment (hanfei1991) [inline proto/analyze.proto]: Where is the Analyze Index Response ?

Comment (lamxTyler) [reply proto/analyze.proto]: Added at line 60.

**Comment by coocood** [comment]
LGTM

**Pull request merged**